modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/LeaseUpdater.java

```diff
@@ -395,4 +407,54 @@ private boolean isLeaseOutdated(Lease lease) {
             return now.after(lease.getExpirationTime());
         }
     }
+
+    /**
+     * Message handler to process notification from replica side.
+     */
+    private class PlacementDriverActorMessageHandler implements NetworkMessageHandler {
+        @Override
+        public void onReceived(NetworkMessage msg0, String sender, @Nullable Long correlationId) {
+            if (!(msg0 instanceof PlacementDriverActorMessage)) {
+                return;
+            }
+
+            var msg = (PlacementDriverActorMessage) msg0;
+
+            if (!stateActorLock.enterBusy()) {
+                return;
+            }
+
+            try {
+                processMessageInternal(sender, msg);
+            } finally {
+                stateActorLock.leaveBusy();
+            }
+        }
+
+        /**
+         * Processes an placement driver actor message. The method should be invoked under state lock.
+         *
+         * @param sender Sender node name.
+         * @param msg Message.
+         */
+        private void processMessageInternal(String sender, PlacementDriverActorMessage msg) {
+            ReplicationGroupId grpId = msg.groupId();
+
+            Lease lease = leaseTracker.getLease(grpId);
+
+            if (msg instanceof StopLeaseProlongationMessage) {
+                if (lease.isProlong() && sender.equals(lease.getLeaseholder().name())) {
```

Comment:
maybe rename this method to `isProlongable`?

Reply:
Renamed.